src/Camera.cpp

```diff
@@ -253,30 +256,33 @@ void Camera::computeReproErrAllBoard() {
  */
 void Camera::refineIntrinsicCalibration(int nb_iterations) {
   ceres::Problem problem;
-  double loss = 1;
+  double loss = 1.0;
   LOG_INFO << "Parameters before optimization :: " << this->getCameraMat();
   LOG_INFO << "distortion vector :: " << getDistortionVectorVector();
   for (std::map<int, std::weak_ptr<BoardObs>>::iterator it =
            board_observations_.begin();
        it != board_observations_.end(); ++it) {
     std::shared_ptr<BoardObs> board_obs_ptr = it->second.lock();
-    if (board_obs_ptr->valid_ == true) {
+    if (board_obs_ptr && board_obs_ptr->valid_ == true) {
```

Comment:
for board_obs_ptr->valid_ you faced any problem without it?

Reply:
I only added `if (board_obs_ptr && ...` to check if the pointer exists. In the line before that, we call `.lock()` and it doesn't guarantee the pointer exists as it is a weak pointer and if it is a multi-threaded environment, another thread might have modified/deleted that pointer.